### ISSUE TITLE
win-arm64 installer fails on Windows arm64 machine

### ISSUE THREAD

**Issue by AArnott** [velopack/velopack#85]
The win-x64 installer runs fine on x64 and arm64 Windows machines. But the win-arm64 installer fails on my ARM64 machine:

![image](https://github.com/velopack/velopack/assets/3548/dd74424e-5966-4e00-88eb-efe38bb24b2a)

```
[Window Title]
Setup Error

[Content]
An error has occurred: This application (arm64) does not support your CPU architecture.

[OK]
```

The following is a snippet of the relevant args to the command I used to build the installer:
```
dotnet vpk pack --packDir bin/publish/Release/win-arm64 --runtime win-arm64 --framework net8.0-arm64-desktop
```

Earlier in my iterating with velopack, I packed my win-arm64 program _without_ specifying `--runtime win-arm64` to the `pack` command, and the (presumably win-x64) installer ran fine on my arm64 Windows machine (and so did my win-arm64 program). It's fantastic that you offer a win-arm64 installer, but why do you suppose it isn't working?

Just to frontload resolution of any dispute that I have an arm64 installation of Windows running, here is Task Manager:
![image](https://github.com/velopack/velopack/assets/3548/28704063-0509-42df-a6d8-f37bf0383e8a)

And a sample of processes, some of which are running as arm64:
![image](https://github.com/velopack/velopack/assets/3548/cc66e81e-4dd2-4aaf-b353-1f3171af395c)

**Comment by caesay** [comment]
I don't actually have an arm64 machine for testing, so a lot of the code there (incl the `--framework` arg) is a bit hand wavey. The installer always ships as x86 which runs in WOW on pretty much every version and architecture of Windows ever. If you provide a `--runtime`, the installer will try to check if the local system is compatible before allowing you to install the software.

My best guess is there is a bug in that compatibility check. 

One thing to test would be to run `Setup.exe --log {somePath.log}`, and pretty early on in the log Velopack will print what version of windows & CPU architecture it thinks it's running on. That might help narrow down whether it's a bug in our [OS detection](https://github.com/velopack/velopack/blob/develop/src/Rust/src/windows/os_info.rs) or in [the compatibility check](https://github.com/velopack/velopack/blob/develop/src/Rust/src/windows/util.rs#L294)

**Comment by caesay** [comment]
Also, I really appreciate that you're providing some great testing and feedback. Feel free to pop in our discord if you'd like to chat or ask any questions that might help you get to the finish line a bit quicker.

**Comment by AArnott** [comment]
Interestingly, adding the `-log some.log` switch makes _nothing_ happen. That is, no error appears, no log is written, and 0 is the exit code.

**Comment by AArnott** [comment]
Using `--log` (two hyphens) produces this:

```
13:57:15 [INFO] Starting Velopack Setup (0.0.462-gf8acc97)
13:57:15 [INFO]     Location: "C:\\Users\\andre\\Downloads\\Nerdbank.Zcash.App-ci-win-arm64-Setup (1).exe"
13:57:15 [INFO]     Silent: false
13:57:15 [INFO]     Verbose: false
13:57:15 [INFO]     Log: Some("C:\\temp\\setup.log")
13:57:15 [INFO]     Install To: None
13:57:15 [INFO] OS: Windows 10.0.22621 (Windows 11 Enterprise) [64-bit], Arch=x86_64
13:57:15 [INFO] Reading bundle header...
13:57:15 [INFO] Bundle offset = 2658816, length = 32085430
13:57:15 [INFO] Loading bundle from embedded zip...
13:57:15 [INFO] Bundle loaded successfully.
13:57:15 [INFO] Reading package manifest...
13:57:15 [INFO] Package manifest loaded successfully.
13:57:15 [INFO]     Package ID: Nerdbank.Zcash.App
13:57:15 [INFO]     Package Version: 0.2.167-beta.g9a08d51f42
13:57:15 [INFO]     Package Title: eZcash
13:57:15 [INFO]     Package Authors: Nerdbank
13:57:15 [INFO]     Package Description: eZcash
13:57:15 [INFO]     Package Machine Architecture: arm64
13:57:15 [INFO]     Package Runtime Dependencies: net8-arm64-desktop
13:57:15 [INFO] Attempting to open global system mutex: 'velopack-Nerdbank.Zcash.App'
13:57:15 [INFO] Checking application pre-requisites...
13:57:15 [INFO]     .NET 8.0.0 Arm64 WindowsDesktop is already installed.
13:57:15 [INFO] Determining install directory...
13:57:15 [INFO] Installation Directory: "C:\\Users\\andre\\AppData\\Local\\Nerdbank.Zcash.App"
13:57:15 [INFO] There is 194.43 Gbit free space available at destination, this package requires 153.33 Mbit.
13:57:15 [ERROR] An error has occurred: This application (arm64) does not support your CPU architecture.
```

**Comment by caesay** [comment]
The issue is that [GetNativeSystemInfo](https://discord.com/channels/767856501477343282/767856501477343286/1228796967094718515) is virtualised on Arm64. We need to use [IsWow64Process2](https://learn.microsoft.com/en-us/windows/win32/api/wow64apiset/nf-wow64apiset-iswow64process2) or possibly even just check the [PROCESSOR_ARCHITEW6432](https://learn.microsoft.com/en-gb/windows/win32/winprog64/wow64-implementation-details) environment variable.

**Comment by caesay** [comment]
I believe this is fixed in ac498494bdbd8ba52d9df3e4ad99b63bc7083393

**Comment by AArnott** [comment]
Confirmed. It's still an x86 process rather than an arm64 process, but the installer succeeds. :) Thanks.
